File: app/src/main/java/com/doubean/ford/data/vo/PostComments.java
package com.doubean.ford.data.vo;

import java.util.List;

public class PostComments {
    private List<PostComment> topComments;
    private List<PostComment> allComments;

    public PostComments() {
    }

    public PostComments(List<PostComment> topComments, List<PostComment> allComments) {
        this.topComments = topComments;
        this.allComments = allComments;
    }

    public List<PostComment> getTopComments() {
        return topComments;
    }

    public void setTopComments(List<PostComment> topComments) {
        this.topComments = topComments;
    }

    public List<PostComment> getAllComments() {
        return allComments;
    }

    public void setAllComments(List<PostComment> allComments) {
        this.allComments = allComments;
    }
}

File: app/src/main/java/com/doubean/ford/model/PostComments.kt
package com.doubean.ford.model

import java.time.LocalDateTime

class PostComments(val topComments: List<PostComment>, val allComments: List<PostComment>)

data class PostComment(
    val id: String,

    val author: User,

    val photos: List<SizedPhoto>?,

    val text: String? = null,

    val created: LocalDateTime?,

    val voteCount: Int,

    val ipLocation: String,

    val repliedTo: RepliedToPostComment? = null,

    )

data class RepliedToPostComment(
    val id: String,

    val author: User,

    val photos: List<SizedPhoto>?,

    val text: String? = null,

    val created: LocalDateTime?,

    val voteCount: Int,

    val ipLocation: String,
)